### ISSUE TITLE
SNS gamemode causing lagging for clients because of "mp_weapon_thermite_grenade"

### ISSUE THREAD

**Issue by Strykus** [R2Northstar/NorthstarMods#882]
SNS gamemode causing lagging for clients because of 
`string offhand_weapon = "mp_weapon_thermite_grenade" // offhand weapon`  in  `Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_sns.gnut` file

Some people teleporting, other having their latency significantly increasing if they start mooving faster around the map

Removing "mp_weapon_thermite_grenade" from players loadout by commenting *line 5* and *line 188* in the `_gamemode_sns.gnut` fixing the issue. 
![изображение](https://github.com/user-attachments/assets/5a9eaa93-cfcc-4ecd-a1d7-b95c47309481)

**Comment by ASpoonPlaysGames** [comment]
This is *very strange* and tbh I have no idea how the two things could be related.

**Comment by ASpoonPlaysGames** [comment]
In fact, how does commenting this line even compile?

**Comment by EM4Volts** [comment]
> This is _very strange_ and tbh I have no idea how the two things could be related.

they really shouldnt?
the weapon doesnt really have something that causes lag without it being heavily modified.

@Strykus are you the only one with this problem or did others confirm this too?
could be another thing like the monarch arc fx

**Comment by Strykus** [comment]
> are you the only one with this problem or did others confirm this too? could be another thing like the monarch arc fx

No. it not only me. TheKraberGuy is the one who tell me when he played on my SNS server that its always the problem with SNS and he have jumping latency from 60 to 100+. Other two people having just laggy gameplay like they have high latency even in the tab its stable 60+- and i myself joined `fvnkhead's sticks` SNS server and having same issue like those two other people.

**Comment by Strykus** [comment]
![изображение](https://github.com/user-attachments/assets/212c73a1-d209-4ce7-9e1e-49ca23113b09)

**Comment by Gazyi** [comment]
[https://github.com/R2Northstar/NorthstarMods/blob/main/Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_sns.gnut#L144](_gamemode_sns.gnut#L144)
Why does this function use string from struct? It doesn't look like modifiable by gamemode rules or cvars. 
Replace it with: `player.GiveOffhandWeapon( "mp_weapon_thermite_grenade", OFFHAND_RIGHT )` and try to run server.
